getSimulationPA.m

```diff
@@ -49,15 +50,14 @@
         if(nderiv>0)
             switch Model.exp{s}.scale
                 case 'log'
-                    nt = size(SP.dmydxi,1);
-                    np = size(SP.dmydxi,4);
-                    ny = size(SP.dmydxi,2);
-                    % To be checked!
+                    nt = size(SP.dCydxi,1);
+                    np = size(SP.dCydxi,4);
+                    ny = size(SP.dCydxi,2);
                     dtmpdxi = arrayfun(@(x,y) diag(squeeze(SP.dCydxi(x,:,:,y))),repmat(1:nt,[np,1]),...
                         repmat(transpose(1:np),[1,nt]),'UniformOutput',false);
-                    dmydxi = bsxfun(@times,my,SP.dmydxi) ...
-                        + bsxfun(@times,my,permute(reshape([dtmpdxi{:}]/2,...
-                        [ny,np,nt]),[3,1,2]));
+                    dmydxi = [dmydxi; bsxfun(@times,my_new,SP.dmydxi) ...
+                        + bsxfun(@times,my_new,permute(reshape([dtmpdxi{:}]/2,...
```

Comment:
Also correct... As ana analogy, in 1D, with mean `y` and variance `v`, we would have:
`d(exp(y + v^2/2))/dxi = exp(y + v^2/2) * (dy/dxi + v * dv/dxi)`